rascal-textmate-core/src/main/rascal/lang/textmate/Conversion.rsc

```diff
@@ -135,12 +133,13 @@ list[ConversionUnit] analyze(RscGrammar rsc) {
 
     // Return
     bool isEmptyProd(prod(_, [\alt(alternatives)], _)) = alternatives == {};
-    list[ConversionUnit] units
-        = [unit(rsc, p, hasNewline(rsc, p), getOuterDelimiterPair(rsc, p), getInnerDelimiterPair(rsc, p, getOnlyFirst = true)) | p <- prods]
-        + [unit(rsc, p, false, <nothing(), nothing()>, <nothing(), nothing()>) | p <- prodsDelimiters, !isEmptyProd(p)]
-        + [unit(rsc, p, false, <nothing(), nothing()>, <nothing(), nothing()>) | p <- prodsKeywords, !isEmptyProd(p)];
+    set[ConversionUnit] units
+        = {unit(rsc, p, false, hasNewline(rsc, p), getOuterDelimiterPair(rsc, p), getInnerDelimiterPair(rsc, p, getOnlyFirst = true)) | p <- prodsNonRecursive}
+        + {unit(rsc, p, true, hasNewline(rsc, p), getOuterDelimiterPair(rsc, p), getInnerDelimiterPair(rsc, p, getOnlyFirst = true)) | p <- prodsRecursive}
+        + {unit(rsc, p, false, false, <nothing(), nothing()>, <nothing(), nothing()>) | p <- prodsDelimiters, !isEmptyProd(p)}
+        + {unit(rsc, p, false, false, <nothing(), nothing()>, <nothing(), nothing()>) | p <- prodsKeywords, !isEmptyProd(p)};
 
```

Comment:
The diff doesn't show it nicely, but the differences on these lines are:
  - the addition of the third parameter in each `unit(...)` constructor;
  - the split of the first set into two (`prods` vs. `prodsNonRecursive` + `prodsRecursive`).

Reply:
a pity those functions have to be repeated, just for the third argument to flip from false to true. there is a way to make the generator run over a tuple that contains the boolean flag (so like `({false} * prodsNonRecursive) + ({true} * prodsRecursive)`. Unsure what I would prefer though.

Reply:
I simplified this code a bit more using a helper function